### PR TITLE
Issue #3516 case insensitive login

### PR DESCRIPTION
**Pull request by megasaturnv**
Pull request for ZoneMinder/zoneminder#3516 to add the option to match mixed case usernames to lower case usernames in database. Useful for LDAP auth which is case-insensitive. Has been compiled and tested

Initially created option `ZM_AUTH_CASE_INSENSITIVE_USERNAMES` but this was too long, so renamed to `ZM_CASE_INSENSITIVE_USERNAMES`. Maintainers may wish to change where this goes in the options list, since it no longer begins with AUTH_

Feel free to write a better description for the docs. Translations may also be wanted for `web/lang/es_la.php` and `web/lang/ru_ru.php` as these have descriptions for similar options

### CODE REVIEW

File: web/includes/auth.php [hunk 1]
```diff
@@ -60,7 +60,12 @@ function migrateHash($user, $pass) {
 function validateUser($username='', $password='') {
   $sql = 'SELECT * FROM Users WHERE Enabled=1 AND Username=?';
   // local user, shouldn't affect the global user
-  $user = dbFetchOne($sql, NULL, array($username));
+  $user = null; // Not global
+  if (ZM_CASE_INSENSITIVE_USERNAMES) {
+      $user = dbFetchOne($sql, NULL, array(strtolower($username)));
+  } else {
+      $user = dbFetchOne($sql, NULL, array($username));
+  }
```
Comment (connortechnology) [inline web/includes/auth.php:65]: This requires the value stored in the db to be lowercase as well.  I think we should alter the sql as well to use tolower(Username).

**Comment by connortechnology** [comment]
Looks good, just a couple comments.  I don't think we should say that this is for AUTH_REMOTE.  It applies to all auth really.  It's just a choice to be case insensitive on usernames.  A lot of people use their actual names as usernames with regular auth and of course they have to type it in case correct.  I think this is a great feature in general.I would actually prefer the config name to start with AUTH but as you say it gets too long so I don't really care.

**Comment by megasaturnv** [comment]
Agreed, didn't feel right to me either only matching against lower case usernames in the database.

As I was changing the code, I realised the readability was much better if the tolower() was done entirely in SQL, rather than one half in PHP and the other in SQL. Makes covering all login options easier too.

Currently testing a build of this on my end

**Comment by megasaturnv** [comment]
No problems with my build from source here. If there are two users with the same name but different cases (e.g. Test and TEst), it will login with the first user with a matching name in the table.

One quirk of PHP - ```If``` statements in don't have their own scope. Let me know if you'd prefer me to add:
```
$sql = null;
```
before each:
```
if (ZM_CASE_INSENSITIVE_USERNAMES) {
  $sql = 'SELECT * FROM Users WHERE Enabled=1 AND LOWER(Username)=LOWER(?)';
} else {
  $sql = 'SELECT * FROM Users WHERE Enabled=1 AND Username=?';
}
```

**Pull request merged**